engine/src/main/java/org/hibernate/search/engine/cfg/impl/EngineConfigurationUtils.java

```diff
@@ -38,15 +38,12 @@ public static ConfigurationPropertySourceExtractor extractorForBackend(Optional<
 		}
 	}
 
-	@SuppressWarnings("deprecation")
 	public static ConfigurationPropertySourceExtractor extractorForIndex(
 			ConfigurationPropertySourceExtractor extractorForBackend, String indexName) {
 		return engineSource -> {
 			ConfigurationPropertySource backendSource = extractorForBackend.extract( engineSource );
 			return backendSource.withMask( BackendSettings.INDEXES ).withMask( indexName )
-					.withFallback( backendSource )
-					.withFallback( backendSource.withMask( BackendSettings.INDEX_DEFAULTS )
-							.onGet( log::deprecatedIndexDefaultsPrefix ) );
```

Comment:
Looks like you're fixing https://hibernate.atlassian.net/browse/HSEARCH-4071 already. Fine by me, but maybe change the ticket key in the commit message?